internal/terraform/context_plan.go

```diff
@@ -201,6 +203,47 @@ The -target option is not for routine use, and is provided only for exceptional
 	// basis to try to give the UI some extra context to return alongside the
 	// error messages.
 
+	// If we have a plan with changes, check if any of the targeted resources are
+	// not among the planned changes, likely due to a typo in resource address.
+	if plan != nil && plan.Changes != nil && len(opts.Targets) > 0 {
+		// Build up a map of changed resources. We will check for the presence of
+		resourceMap := make(map[string]struct{}, len(plan.Changes.Resources))
+		for _, resource := range plan.Changes.Resources {
+			resourceMap[resource.Addr.String()] = struct{}{}
+		}
+
+		// We will build up suggestions if necessary based on previous state only once
+		var once sync.Once
+		var suggestions []string
+
+		for _, target := range opts.Targets {
+			if _, ok := resourceMap[target.String()]; ok {
+				continue
+			}
+
+			once.Do(func() {
+				if plan.PrevRunState == nil {
+					return
+				}
+				for _, resource := range plan.PrevRunState.AllResourceInstanceObjectAddrs() {
+					suggestions = append(suggestions, resource.Instance.Resource.Resource.String())
+				}
+			})
+
+			suggestion := didyoumean.NameSuggestion(target.String(), suggestions)
```

Comment:
I see the func comment here suggests that we use a relatively small number (less than hundreds). I can update the code to only pass a slice of X if it is longer. After all, even in the case that we remove resource number 189 that would have been suggested, we'll still prompt with "Check for a typo."